core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java

```diff
@@ -630,7 +631,11 @@ private String getRawMessage(IMessageBrowser<?> messageBrowser, IListener listen
 			} else {
 				try {
 					if (listener!=null) {
-						msg = listener.extractMessage(rawmsg, null).asString();
+						if(listener instanceof JavaListener) { // TODO: remove this when JavaListener uses Message
+							msg = listener.extractMessage(Message.asString(rawmsg), null).asString();
+						} else {
+							msg = listener.extractMessage(rawmsg, null).asString();
+						}
```

Comment:
This change should not be necessary, as messageBrowser returns either MessageWrapper, or M of Listener<M>